src/loader.rs

```diff
@@ -1,4 +1,9 @@
 use super::*;
+use std::collections::HashSet;
+
+// This regex defines the syntax for a Justfile include statement as `!include <file-path>`
+// occurring at the start of a line, and as the only contents of that line
+const INCLUDE_REGEX: &str = "^!include\\s+(.+)$";
```

Comment:
I think we should match `!include` on its own:

```suggestion
const INCLUDE_REGEX: &str = "^!include(\\s(.+))?$";
```

Which would be an error.

Reply:
Actually, on second thought, can we can do more work with this regex? Like using it to extract all comments and includes before the first non-blank, non-comment line?

```suggestion
const DIRECTIVE_REGEX: &str = r"(?x)
  (
    ^\s*$ # blank
    ^\#.*$ # comment
    ^(<?P<directive>!.*)$ # directive
    (\n|\r\n)
)*
";
```

This could extract all leading directives, which we could then process individually. All directives other than `!include` are an error.

Reply:
I'm reluctant to have this regex do too much work, since it involves codifying the syntax of a valid justfile in multiple places that can get out of sync. Really, I think that using a regex here to do preliminary parsing of one syntactic construct is a quick hack, and all the code that deals with parsing a justfile really ought to live in the parsing module, and at some point we should do the fairly-hefty redesign of having the parsing step take place before includes (or module imports). In the meantime, hopefully any further directives that prove necessary can happen in the parser and don't need to happen in the loader step.